### PR TITLE
feat(planning): Add the option to upload files for planning items (SDN…

### PR DESCRIPTION
**Pull request by ioanpocol**
…TB-529)

### CODE REVIEW

File: client/components/Planning/PlanningEditor/index.jsx [hunk 1]
```diff
@@ -568,6 +603,30 @@ export class PlanningEditorComponent extends React.Component {
                             onFocus={onFocusDetails}
                         />}
                     </ToggleBox>
+
+                    <ToggleBox
```
Comment (MarkLark86) [inline client/components/Planning/PlanningEditor/index.jsx]: Please use the `planning_types` collection to control if `Files` are to be available for `Planning Items`

Comment (ioanpocol) [reply client/components/Planning/PlanningEditor/index.jsx]: fixed

**Comment by ioanpocol** [comment]
@MarkLark86 the requested changes are done
@nrvikas 1. is done, working on 2.

**Comment by ioanpocol** [comment]
@nrvikas 2. is done too

**Comment by MarkLark86** [comment]
@ioanpocol Looks good except for one thing. The `PlanningEditor` uses `actions.events.api.uploadFiles` to upload the files, which will hit the `event_files` endpoint and not your new `planning_files` endpoint.

If for some reason a user is given `planning_planning_management` privilege and not `planning_event_management` privilege then that user will be able to create `Planning Items` but not be able to upload files. Keeping it as a separate endpoint is good to keep the Planning & Event privileges separate.

Might be good to just create a generic `uploadFiles` action and pass in the endpoint url, i.e. `events_files` or `planning_files`.

**Comment by ioanpocol** [comment]
@MarkLark86 Good catch, thank you. Done.

**Comment by MarkLark86** [comment]
@ioanpocol `PlanningEditor` is using `events_files` to remove the file:
`removeFile: (file) => dispatch(actions.events.api.removeFile(file)),`.

Looks like that's the last one. After that change, please squash your commits and this one is good to merge.
:thumbsup:

**Comment by ioanpocol** [comment]
Done

**Pull request merged**